src/sentry/models/orgauthtoken.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.encoding import force_text
+
+from sentry.conf.server import SENTRY_SCOPES
+from sentry.db.models import (
+    ArrayField,
+    BaseManager,
+    FlexibleForeignKey,
+    Model,
+    control_silo_only_model,
+    sane_repr,
+)
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+def validate_scope_list(value):
+    for choice in value:
+        if choice not in SENTRY_SCOPES:
+            raise ValidationError(f"{choice} is not a valid scope.")
+
+
+@control_silo_only_model
+class OrgAuthToken(Model):
+    __include_in_export__ = True
+
+    organization = HybridCloudForeignKey("sentry.Organization", null=False, on_delete="CASCADE")
+    # The JWT token in hashed form
+    token_hashed = models.CharField(max_length=3000, unique=True, null=False)
+    # An optional representation of the last characters of the original token, to be shown to the user
+    token_last_characters = models.CharField(max_length=4, null=True)
+    name = models.CharField(max_length=255, null=False)
+    scope_list = ArrayField(
+        models.TextField(),
+        validators=[validate_scope_list],
+    )
+
+    created_by = FlexibleForeignKey("sentry.User", null=True, blank=True, on_delete="SET_NULL")
+    date_added = models.DateTimeField(default=timezone.now, null=False)
+    date_last_used = models.DateTimeField(null=True, blank=True)
+    project_last_used = HybridCloudForeignKey(
+        "sentry.Project", null=True, blank=True, on_delete="SET_NULL"
+    )
+
+    objects = BaseManager(cache_fields=("token",))
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_orgauthtoken"
+
+    __repr__ = sane_repr("organization_id", "token")
```

Comment:
Hmm, mypy is complaining here:

```
error: Incompatible types in assignment (expression has type "Callable[[Model], str]", base class "object" defined the type as "Callable[[object], str]")  [assignment]
```

Not sure if/what I can/should do about this...? Seems correct to me, and same way this is used everywhere else...?

Reply:
Note: I added this to the ignore list to stop mypy from complaining.